src/rust/engine/fs/src/directory.rs

```diff
@@ -671,7 +661,7 @@ impl DigestTrie {
                 Entry::Directory(d) => {
                     f(&path, entry);
                     d.tree
-                        .walk_helper(root, path.to_path_buf(), symlink_behavior, link_depth, f);
+                        .walk_helper(root, path.to_path_buf(), symlink_behavior, link_depth + 1, f);
```

Comment:
I think the `link_depth += 1` call should be retained where it was, as I think it's meant to be counting the number of links traversed, not the depth of the walk.

Reply:
Done